climada/engine/unsequa/calc_base.py

```diff
@@ -500,10 +536,44 @@ def _calc_sens_df(method, problem_sa, sensitivity_kwargs, param_labels, X, unc_d
         else:
             sens_indices = method.analyze(problem_sa, Y,
                                                     **sensitivity_kwargs)
+        #refactor incoherent SALib output
+        nparams = len(param_labels)
+        if method.__name__ == 'SALib.analyze.ff':
+            if sensitivity_kwargs['second_order']:
+                #parse interaction terms of sens_indices to a square matrix
+                #to ensure consistency with unsequa
+                interaction_names = sens_indices.pop('interaction_names')
+                interactions = np.full((nparams, nparams), np.nan)
+                for i,ie in enumerate(interaction_names):
+                    interactions[param_labels.index(ie[0]),
+                                 param_labels.index(ie[1])] = sens_indices['IE'][i]
+                sens_indices['IE'] = interactions
+
+        if method.__name__ == 'SALib.analyze.hdmr':
+            keys_to_remove = ['Em','Term','X','Y'] #need to delete emulator as is a dict
+                                                #and other useless variables
+            [keys_to_remove.append(si)  for si, si_val_array in sens_indices.items()
+             if np.array(si_val_array).size > nparams**2]
+            sens_indices = {k: v for k, v in sens_indices.items()
```

Comment:
There seems to be a space too much.

Reply:
Do you mean in front of the if? It was added automatically by VScode..